go/database/mpt/proof_test.go

```diff
@@ -419,6 +420,57 @@ func TestWitnessProof_String(t *testing.T) {
 	}
 }
 
+func TestWitnessProof_Is_Valid(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	address := common.Address{0xAB, 0xCD, 0xEF}
+
+	ctxt := newNodeContextWithConfig(t, ctrl, S5LiveConfig)
+
+	desc := &Extension{
+		path: AddressToNibblePath(address, ctxt)[0:30],
+		next: &Account{address: address, pathLength: 10, info: AccountInfo{Nonce: common.Nonce{0x01}, Balance: common.Balance{0x02}, CodeHash: common.Hash{0x03}}},
+	}
+
+	root, node := ctxt.Build(desc)
+	totalProof := createReferenceProof(t, ctxt, &root, node)
+	handle := node.GetReadHandle()
+	defer handle.Release()
+	rootHash, dirty := handle.Get().GetHash()
```

Comment:
nit: reading the hash technically requires View access, but here it does not matter.

Reply:
ah, sorry for repeating the same mistake. Anyway, this code was removed, since the rootHash is not needed anymore